app/webroot/js/generic_functions.js

```diff
@@ -85,3 +91,20 @@ $(document).ready(function() {
         key_navigation();
     }
 });
+
+/**
+ * Changes Flag in Add Translation box on Changing Language
+ */
+
+$(document).ready(function() {
+    $( ".language-selector" ).change(function() {
+        if($(this).val()!='auto'){
+            var lang_flag_url = '/img/flags/'+$(this).val()+'.png';
+            $(this).next().attr('src',lang_flag_url);
+            $(this).next().attr('alt',$(this).val());
+        }else{
+            $(this).next().attr('src','/img/flags/unknown.png');
+            $(this).next().attr('alt','Unknown');
+        }
+    });
+});
```

Comment:
I think this Javascript code should actually go into `app/webroot/js/sentences.add_translation.js` because it isn’t used in all the pages, only in pages that include translatable sentences.